compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1107,6 +1154,135 @@ public void visit(BLangSimpleVariable varNode) {
         env.enclPkg.isListenerAvailable |= Symbols.isFlagOn(varNode.symbol.flags, Flags.LISTENER);
     }
 
+    @Override
+    public void visit(BLangUserDefinedType userDefinedType) {
+        if (Types.getReferredType(userDefinedType.getBType()).getKind() == TypeKind.RECORD) {
+            createNewTypeDescInst(userDefinedType.getBType());
+        }
+    }
+
+    @Override
+    public void visit(BLangTupleTypeNode tupleTypeNode) {
+        createNewTypeDescInst(tupleTypeNode.getBType());
+        BLangType typeNode;
+        for (BLangSimpleVariable member : tupleTypeNode.memberTypeNodes) {
+            typeNode = member.typeNode;
+            if (member.typeNode != null) {
+                typeNode.accept(this);
+            }
+        }
+        typeNode = tupleTypeNode.restParamType;
+        if (typeNode != null) {
+            typeNode.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(BLangValueType valueType) {
+    }
+
+    @Override
+    public void visit(BLangUnionTypeNode unionTypeNode) {
+        unionTypeNode.memberTypeNodes.forEach(typeNode -> typeNode.accept(this));
+    }
+
+    @Override
+    public void visit(BLangRecordTypeNode recordTypeNode) {
+        createNewTypeDescInst(recordTypeNode.getBType());
+        for (BLangSimpleVariable field : recordTypeNode.fields) {
+            if (field.typeNode != null) {
+                field.typeNode.accept(this);
```

Comment:
Codecov warnings. Fields of record types are a common scenario too.

Reply:
In the desugaring phase, we created a user-defined type for the local record. I have now changed it to retain it as a record-type node without changing it to a user-defined type node

Reply:
Earlier, in the desugaring phase, we created a user-defined type for the local record. I have now changed it to retain it as a record type without changing it to a user-defined type.